bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -165,31 +154,31 @@ public AbstractReconnectStrategy getReconnectStrategy() {
     }
 
     /**
-     * @return name for the connection as defined in smarthome.cfg.
+     * Get the MQTT broker host
      */
-    public String getName() {
-        return name;
+    @NonNull
+    public String getHost() {
+        return host;
     }
 
     /**
-     * Get the url for the MQTT broker. Valid URL's are in the format:
-     * tcp://localhost:1883 or ssl://localhost:8883
-     *
-     * @return url for the MQTT broker.
+     * Get the MQTT broker port
      */
-    public String getUrl() {
-        return url;
+    @Nullable
+    public Integer getPort() {
+        return port;
```

Comment:
On the constructor you state.
```text
* @param port A port or null to select the default port for a secure or insecure connection
```

So I would assume that a port is choosen now and we could return the explicit set one, the insecure or secure default.
I am pretty sure that a connection uses a port ;)